classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php

```diff
@@ -72,43 +72,39 @@ public function up() {
 
 		// pkp/pkp-lib#6093 Don't allow nulls (previously an upgrade workaround)
 		Capsule::schema()->table('user_settings', function (Blueprint $table) {
-			$table->bigInteger('assoc_type')->default(0)->change();
-			$table->bigInteger('assoc_id')->default(0)->change();
+			$table->bigInteger('assoc_type')->default(0)->nullable(false)->change();
+			$table->bigInteger('assoc_id')->default(0)->nullable(false)->change();
 		});
 		Capsule::schema()->table('announcement_types', function (Blueprint $table) {
-			$table->bigInteger('assoc_type')->change();
+			$table->bigInteger('assoc_type')->nullable(false)->change();
 		});
 		Capsule::schema()->table('email_templates', function (Blueprint $table) {
-			$table->bigInteger('context_id')->default(0)->change();
+			$table->bigInteger('context_id')->default(0)->nullable(false)->change();
 		});
 		Capsule::schema()->table('genres', function (Blueprint $table) {
-			$table->bigInteger('seq')->change();
-			$table->smallInteger('supplementary')->default(0)->change();
+			$table->bigInteger('seq')->nullable(false)->change();
+			$table->smallInteger('supplementary')->default(0)->nullable(false)->change();
 		});
 		Capsule::schema()->table('event_log', function (Blueprint $table) {
-			$table->bigInteger('assoc_type')->change();
-			$table->bigInteger('assoc_id')->change();
+			$table->bigInteger('assoc_type')->nullable(false)->change();
+			$table->bigInteger('assoc_id')->nullable(false)->change();
 		});
 		Capsule::schema()->table('email_log', function (Blueprint $table) {
-			$table->bigInteger('assoc_type')->change();
-			$table->bigInteger('assoc_id')->change();
+			$table->bigInteger('assoc_type')->nullable(false)->change();
+			$table->bigInteger('assoc_id')->nullable(false)->change();
 		});
 		Capsule::schema()->table('notes', function (Blueprint $table) {
-			$table->bigInteger('assoc_type')->change();
-			$table->bigInteger('assoc_id')->change();
-		});
-		Capsule::schema()->table('review_forms', function (Blueprint $table) {
-			$table->bigInteger('assoc_type')->change();
-			$table->bigInteger('assoc_id')->change();
+			$table->bigInteger('assoc_type')->nullable(false)->change();
+			$table->bigInteger('assoc_id')->nullable(false)->change();
 		});
 		Capsule::schema()->table('review_assignments', function (Blueprint $table) {
-			$table->bigInteger('review_round_id')->change();
+			$table->bigInteger('review_round_id')->nullable(false)->change();
 		});
 		Capsule::schema()->table('authors', function (Blueprint $table) {
-			$table->bigInteger('publication_id')->change();
+			$table->bigInteger('publication_id')->nullable(false)->change();
 		});
 		Capsule::schema()->table('edit_decisions', function (Blueprint $table) {
-			$table->bigInteger('review_round_id')->change();
+			$table->bigInteger('review_round_id')->nullable(false)->change();
```

Comment:
I believe this should still be `NULL`able for decisions that aren't related to the review stages.

Reply:
@asmecher, are you sure on this one? On my test instances 3.2.0 and 3.2.1, those (that aren't associated with the review stage) have a `0` value. On the master branch, `accept and skip review` records the following:
```
edit_decision_id	submission_id	review_round_id	stage_id	round	editor_id	decision	date_decided		
566	254	0	1	0	1	1	2020-11-24 11:20:06	
```
Or this is going to be something new? In this case all those values should be converted from `0` to `null` as per previous comment.

Reply:
You're right, currently it defaults to `0`. In the future we'll need to convert that to `NULL`, but for now, it can stay as `0`.

Reply:
If it's planned to make null values there, I've removed those lines completely from the script (the last commit).

Reply:
OK, but you'd need to permit `NULL` values here: https://github.com/pkp/pkp-lib/pull/6347/files#diff-f4951b8d90b205bf60a1111a761b62eb75f598c0c536738da01d027561a33f1cR101
...and, for consistency (so we don't have both `NULL` and `0` values), remove the `(int)` casts where review assignments are inserted and updated. (Use `DAO::nullOrInt` instead.)

Reply:
@asmecher, I've updated PRs but I don't get it. How it's possible that review assignments may have null or int values for review round ID? Those can be inserted or updated outside of the review process, like editor decisions?

Reply:
Sorry, @Vitaliy-1, I'm yet again reading too quickly and leading you astray. I had meant the `edit_decisions` table but you've already removed that line and I was pointing at the wrong table.

Reply:
No problem. I've removed the last commit.